src/sql/ResourceSelector.tsx

```diff
@@ -18,6 +18,7 @@ export type ResourceSelectorProps = {
   // Options only needed for the ConfigEditor
   title?: string;
   disabled?: boolean;
+  allowCustomValue?: boolean;
```

Comment:
can we make `ResourceSelectorProps` an extension of the options available for the Select? That way we can remove some of the options here and make it more extensible. I tried `export interface ResourceSelectorProps extends SelectCommonProps<string>` and seemed to work here.

This approach cannot be used with `ConfigSelectProps` though since that already extends `DataSourcePluginOptionsEditorProps` and have some conflicting properties (in particular the property `options`) but maybe we can expose more options there so we don't need to extend this for a while. Checking the properties of the Selector I think we can add some common ones:

```
    allowCustomValue?: boolean;
    autoFocus?: boolean;
    backspaceRemovesValue?: boolean;
    className?: string;
    disabled?: boolean;
    invalid?: boolean;
    isClearable?: boolean;
    isMulti?: boolean;
    inputId?: string;
    showAllSelectedWhenOpen?: boolean;
    maxMenuHeight?: number;
    minMenuHeight?: number;
    maxVisibleValues?: number;
    menuPlacement?: 'auto' | 'bottom' | 'top';
    menuPosition?: 'fixed' | 'absolute';
    noOptionsMessage?: string;
    onBlur?: () => void;
    onCreateOption?: (value: string) => void;
    onInputChange?: (value: string, actionMeta: InputActionMeta) => void;
    placeholder?: string;
    width?: number;
    isOptionDisabled?: () => boolean;
```

Reply:
yes, sounds good 👍